lib/views/item-scene.js

```diff
@@ -22,6 +22,49 @@ class ItemScene extends React.Component {
   constructor(props) {
     super(props);
     debug('created');
+
+    let distance = this.props.item.distance;
+    if (typeof distance !== 'number') {
+      distance = null;
+    }
+
+    this.state = {
+      distance,
+    };
+  }
+
+  componentDidMount() {
+    if (this.state.distance !== null ||
+      typeof this.props.item.latitude !== 'number' ||
+      typeof this.props.item.longitude !== 'number') {
+      return;
+    }
+
+    navigator.geolocation.getCurrentPosition((pos) => {
```

Comment:
As the users current location is 'state' I think this should be part of our 'atomic' Redux store. I suggest that we have a central method somewhere that `startLocationUpdates()` and `stopLocationUpdates()` which internally calls a redux action `.locationChange(lat, lon)`.

When the `.locationChange()` action is called it can set the lat/lon on the global state object, and any component that depend on that state can update.